JavaScript/JavaScriptSDK/SendBuffer.ts

```diff
@@ -132,6 +136,17 @@ module Microsoft.ApplicationInsights {
         }
 
         public enqueue(payload: string) {
+            if (this._buffer.length >= SessionStorageSendBuffer.MAX_BUFFER_SIZE) {
+                // sent internal log only once per page view
+                if (!this._bufferFullMessageSent) {
+                    _InternalLogging.throwInternalUserActionable(LoggingSeverity.WARNING,
+                        new _InternalLogMessage(_InternalMessageId.USRACT_SessionStorageBufferFull,
+                            "Maximum buffer size reached: " + this._buffer.length));
+                    this._bufferFullMessageSent = true;
```

Comment:
Good point. It will be cleared when the script is loaded again (when user navigates to a next page, etc). I didn't want to log this message too often... but you're right, it might be a good idea to clear it when the buffer has space again.